src/test_tri_vert_orientation.cpp

```diff
@@ -0,0 +1,57 @@
+#include <Omega_h_macros.h>
+
+#include <Omega_h_adj.hpp>
+#include <Omega_h_for.hpp>
+#include <Omega_h_mesh.hpp>
+#include <cstdio>
+#include <Omega_h_file.hpp>
+
+bool is_ccw_oriented(Omega_h::Mesh& mesh) {
+  OMEGA_H_CHECK(mesh.dim() == 2);
+  bool ccw;
+  // find mesh orientation: node number orientation of faces
+  // clockwise or counter clockwise: 1 for clockwise, -1 for counter clockwise
+  const auto& face2nodes = mesh.ask_down(Omega_h::FACE, Omega_h::VERT).ab2b;
+  const auto& coords = mesh.coords();
+  Omega_h::Write<Omega_h::LO> ccw_face(mesh.nfaces(), 0, "ccw_face");
+
+  auto find_ccw_face = OMEGA_H_LAMBDA(const Omega_h::LO face) {
+    const auto faceVerts = Omega_h::gather_verts<3>(face2nodes, face);
+    const auto faceCoords = Omega_h::gather_vectors<3, 2>(coords, faceVerts);
+    // calculate using (y2 - y1) * (x3 - x2) - (y3 - y2) * (x2 - x1)
+    const bool cw = (faceCoords[1][1] - faceCoords[0][1]) *
+                            (faceCoords[2][0] - faceCoords[1][0]) -
+                        (faceCoords[2][1] - faceCoords[1][1]) *
+                            (faceCoords[1][0] - faceCoords[0][0]) >
+                    0;
+    ccw_face[face] = (!cw) ? 1 : -1;
+  };
+  Omega_h::parallel_for(mesh.nfaces(), find_ccw_face, "find_ccw_face");
```

Comment:
If the intention is to simply answer the question "are all faces ccw?" then this can be written as a max reduction and we can avoid storing `ccw_face[face]`.

Reply:
If this check is going into the `Omega_h` constructor I agree.

Reply:
PUMI has a very useful ['verify' function](https://github.com/SCOREC/core/blob/24dd23e0191ffe0474ca7d952b8c41e802ccbdf1/apf/apfVerify.cc#L800) that checks for mesh and model classification issues (I'm not sure of the full set of checks offhand).  This function seems like a good first step towards an Omega_h version of that.